idaes/surrogate/pysmo/sampling.py

```diff
@@ -828,24 +921,30 @@ class HammersleySampling(SamplingMethods):
 
     """
 
-    def __init__(self, data_input, number_of_samples=None, sampling_type=None):
+    def __init__(self, data_input, number_of_samples=None, sampling_type=None, xlabels=None, ylabels=None):
```

Comment:
To a cursory glance, it looks like most of this code is repeated in each `SamplingMethods` subclass. There might be a legitimate reason for this (e.g. there are intrinsic differences that require separate specific implementations), but if not, it might be worth considering to extract it to a common function or method in the subclass (if this is already done, apologies in advance, I haven't gone through the entire code in detail).

Reply:
Yes, there are minor differences that require slightly different implementations. Thanks for the suggestion though.